sentry/src/access.rs

```diff
@@ -28,21 +28,27 @@ pub async fn check_access(
         Event::Close => true,
         _ => false,
     };
+    let current_time = Utc::now();
+    let is_in_withdraw_period = current_time > channel.spec.withdraw_period_start;
 
-    // Check basic access rules
-    // only the creator can send a CLOSE
+    // We're only sending a CLOSE
+    // That's allowed for the creator normally, and for everyone during the withdraw period
+    if events.iter().all(is_close_event) && (session.uid == channel.creator || is_in_withdraw_period) {
+        return Ok(());
+    }
+
+    // Only the creator can send a CLOSE
     if session.uid != channel.creator && events.iter().any(is_close_event) {
         return Err(Error::OnlyCreatorCanCloseChannel);
     }
 
-    let current_time = Utc::now();
 
     if current_time > channel.valid_until {
```

Comment:
you must move this to the beginning

Reply:
I meant move the expired check: ` if current_time > channel.valid_until`

Reply:
Fixed! You can review it again, I am resolving this comment.